Kulla

kudav opened the pull request:
Various changes, I'll start with feature branches here from now on as well:

beams3d_slow:
- reorganized function
- implemented new collision operators
- option to supply multiple ion species

beams3d_apply_bpert:
- improvements to calculation

beams3d_normdist:
- made version check meaningful

beams3d_from_fidasim:
- to be implemented

fieldlines_follow:
- check for file type (beams3d or fieldlines), and appropriate procedure

plot_fieldlines:
- added option for color coding the traced path lenght
- added keyboard movement

VMECplot:
-  now has STL export of equilibria
- option to show min/max values in 2D plots

isotoro:
- option for specifying file name

read_hdf5:
- fix for reading in subgroup attributes

plot_fidasim:
- Neutral density plots
- birth profiles
- 2D profiles
- supplying indices for toroidal and poloidal slices

get_bes_fida_aug_data.m:
- added option for plotting the background bremsstrahlung profile
- fixed instrument function

plot_fidasim_profiles:
- option for background bremsstrahlung
- specify plotting line style

New functions:
- beams3d_suv2rzp converts points (ie the beams3d distribution grid) to cylindrical coordinates
- beams3d_calc_beamdensity calculates the 3D beam density distribution - should be merged with beams3d_beamdensity

I need to clean up some things before merging, so this is a draft for now.

kudav committed:
Fix to gitignore

kudav committed:
Adressed issue #5 + stable min/max display

kudav committed:
Fix Merge

kudav committed:
Added neutral density plot to plot_fidasim

kudav committed:
Merge remote-tracking branch 'upstream/master' into Kulla

kudav committed:
Raven changes

kudav committed:
Normdist now has version check

kudav committed:
Merge branch 'kudav_tmp' into Kulla

kudav committed:
Merge branch 'Kulla' of https://github.com/kudav/matlabVMEC into Kulla

kudav committed:
PLOT_FIDASIM: added 2d profiles

kudav committed:
beam_slow now consistent for AUG Validation

kudav committed:
Raven Changes

kudav committed:
Merge commit '020abc0013b0df9556efdc09a6fae4140146815f' into Kulla

kudav committed:
AUG routine from raven

kudav committed:
Revert get_aug

kudav committed:
Added index for slices in plot_fidasim

kudav committed:
Slight changes to plot_fieldlines

kudav committed:
Merge branch 'Kulla' of https://github.com/kudav/matlabVMEC into Kulla

kudav committed:
Impurity species for beams3d_slow

kudav committed:
Added birth reading to plot_fidasim

kudav committed:
Merge branch 'Kulla' of github.com:kudav/matlabVMEC into Kulla

kudav committed:
More birth plots for plot_fidasim

kudav committed:
Added background plotting

kudav committed:
Added calibration to get_aug

kudav committed:
additions

kudav committed:
Various updates and fixes

kudav committed:
Work on Raven

kudav committed:
Fix plot_beams conflict

kudav committed:
Work on beams3d_slow

kudav committed:
Implemented actual beams3d slowing calculation

kudav committed:
Modified calculation with zmean

kudav committed:
Re-Organized beams3d_slow

Now has reasonable names for forumlations

kudav committed:
Create beams3d_suv2rzp.m

kudav committed:
Various additions, work on beams3d_slow

kudav committed:
Create beams3d_from_fidasim.m

kudav committed:
work on plot_fidasim

kudav committed:
Merge branch 'Kulla' of github.com:kudav/matlabVMEC into Kulla

kudav committed:
Raven work

kudav committed:
Added suv2rzp

kudav committed:
Fixed read_hdf5 for subgroup attributes

kudav committed:
Merge branch 'Kulla' of https://github.com/kudav/matlabVMEC into Kulla

kudav committed:
Small fixes

kudav committed:
added R birth plots to plot_beams

kudav committed:
Fix read_hdf5

kudav committed:
Merge commit 'da9ea7c620d67459cca2bdce82082ecfafd1d9db' into Kulla

kudav committed:
some changes to plot_fidasim

kudav committed:
Create beams3d_calc_beamdensity.m

kudav committed:
Update plot_fidasim.m

kudav committed:
Create beams3d_calc_srhoxi.m

kudav committed:
Reformat and add mean to ep in plot_fidasim

kudav committed:
Corrected axis labels

kudav committed:
Recalc distribution

kudav committed:
Fixes and renormalization of calc_srhoxi

kudav committed:
Added flag ommitting losses in depo calc

kudav committed:
Srhoxi fix

kudav committed:
time slice plot of spectrum

kudav committed:
Merge branch 'Kulla' of github.com:kudav/matlabVMEC into Kulla

kudav committed:
Fix? for FIDA data

kudav committed:
Various fixes

kudav committed:
Merge branch 'Kulla' of https://github.com/kudav/matlabVMEC into Kulla

kudav committed:
Implemented debug for FIDA data

kudav committed:
Fixed Bugs in ascot5_plotorbits

kudav committed:
apply_bpert formulation only for one n,m

kudav committed:
plot_beams: better calculation of caxis

kudav committed:
plot_fidasim: cosmetic improvements

kudav committed:
plot_fieldlines: added setting axis limits

kudav committed:
pixplot: rounding of tick labels

kudav committed:
raven changes - wvl

kudav committed:
Merge branch 'Kulla' of github.com:kudav/matlabVMEC into Kulla

lazersos commented on BEAMS3D/beams3d_calc_srhoxi.m:
Proper header text with description, date, version, author, and example usage required.

lazersos commented on BEAMS3D/beams3d_from_fidasim.m:
Proper header text with description, date, version, author, and example usage required.

lazersos commented on WALL/plot_wall_acc.m:
This isn't the behavior we want, is it?

lazersos commented on utilities/get_bes_fida_aug_data.m:
AUG specific routines should be in a separate AUG repository like we do for W7-X.

lazersos reviewed (commented):
There are a few points to clarify.  Also we should probably create a separate FIDASIM directory for FIDASIM routines.

kudav committed:
New FIDASIM dir and other file organizations

kudav committed:
Added min_faces option to plot_wall_acc

kudav committed:
Added description to beams3d_calc_srhoxi

kudav reviewed (commented):
Addressed the points and cleaned up the superfluous/unfinished routines.

lazersos commented on BEAMS3D/beams3d_suv2rzp.m:
Needs proper help header.

lazersos replied:
Still needs a 'proper' header like other files.

lazersos commented on BEAMS3D/beams3d_slow.m:
Can you show that this routine returns simliar results to the old routine for the proper selection of log_type?

lazersos replied:
Confirmed.

lazersos reviewed (changes requested):
Still a couple more things to do.

kudav committed:
Added help header to beams3d_suv2rzp

kudav committed:
Cleanup beams3d_slow

lazersos reviewed (changes requested):
Still have to give a proper header and show that slow produces similar results.

kudav committed:
beams3d_slow: old nrl calc, removed legacy opt.

kudav committed:
beams3d_slow: added verbosity flag

kudav committed:
beams3d_slow: lots of changes

Added descriptions for different slowing down formulations
Logic for dealing with single- and multi-species runs is consistent (with checks)
Removed commented-out code and improved namings

kudav committed:
beams3d_suv2rzp: Updated function header

kudav commented:
Here is the heating plot from beams3d_slow with the current commit:
`slow=beams3d_slow(data,'NRL','plots');`

![grafik](https://github.com/lazersos/matlabVMEC/assets/72251199/a61aedf5-bd4b-49e6-a2e9-3dcc41576a97)

and from commit 18707fd7a328bd6a6156583a45ba8ad5bb9638ab (second), which was where we introduced the multiple coulomb logarithms:
`
slow=beams3d_slow(data,'NRLions_logarithm','plots');`

![grafik](https://github.com/lazersos/matlabVMEC/assets/72251199/3d794a05-96e5-4b6e-b9e9-e2f0d40ba507)

The calculation between the two did not change, thus the output is numerically the same.

lazersos reviewed (approved):
Merge. We will add your biological and technological distinctiveness to our own.

lazersos committed:
Merge branch 'master' into Kulla
